pkg/rbac/authorization.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"fmt"
+
+	"strings"
+
+	"github.com/dgrijalva/jwt-go"
+	api "github.com/onosproject/onos-lib-go/api/rbac"
+	"google.golang.org/grpc"
+)
+
+const (
+	// GroupsKey groups key in the token claims
+	GroupsKey = "groups"
+)
+
+// Rbac role based access control interface
+type Rbac interface {
+	Authorize() error
+}
+
+// NewAuthorization creates a new Authorization instance
+func NewAuthorization(claims jwt.MapClaims, info *grpc.UnaryServerInfo) Authorization {
+	return Authorization{
+		claims: claims,
+		info:   info,
+	}
+}
+
+// SetClaims set jwt token claims
+func (a *Authorization) SetClaims(claims jwt.MapClaims) {
```

Comment:
probably not needed, since you have "NewAuthorization", and it's unlikely to change

Reply:
Will be removed.